### PR TITLE
Add support for Resize Bilinear operation

### PR DESCRIPTION
**Pull request by Rjasuja**
Adds support for Resize bilinear operation for FLOAT32 and QUANt8 type

Signed-off-by: Ritul Jasuja <ritul.jasuja@intel.com>

### CODE REVIEW

File: ngraph_creator/operations/src/ResizeBilinear.cpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            // Already forced NCHW, propogate the flag
+            mNgraphNodes->setForcedNchw(mDefaultOutputIndex, true);
+        }
+    } else if (!useNchw) {  // No conversion needed if useNchw set
+        inputNode = transpose(NHWC_NCHW, inputNode);
+        mNgraphNodes->setForcedNchw(mDefaultOutputIndex, true);
+        ALOGD("%s Forced NCHW conversion at operationIndex %d", __func__, mNnapiOperationIndex);
+    }
+
+    std::vector<float> scaleFactors = {1.0f, 1.0f, width_scale, height_scale};
```
Comment (akodanka) [inline ngraph_creator/operations/src/ResizeBilinear.cpp:142]: Shouldn't this be {1.0f, 1.0f, height_scale, width_scale} ?

File: ngraph_creator/operations/src/ResizeBilinear.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            std::make_shared<ngraph::opset3::ShapeOf>(inputNode), ngraph::element::f16);
+        scaleConstNode = createConstNode(ngraph::element::f16, {4}, scaleFactors);
+    } else if (checkInputOperandType(1, (int32_t)OperandType::FLOAT32) ||
+               checkInputOperandType(1, (int32_t)OperandType::INT32)) {
+        shape = std::make_shared<ngraph::opset3::Convert>(
+            std::make_shared<ngraph::opset3::ShapeOf>(inputNode), ngraph::element::f32);
+        scaleConstNode = createConstNode(ngraph::element::f32, {4}, scaleFactors);
+    }
+    // Scale input shape to calculate final output shape
+    auto outputShape = std::make_shared<ngraph::op::v1::Multiply>(scaleConstNode, shape);
```
Comment (akodanka) [inline ngraph_creator/operations/src/ResizeBilinear.cpp:157]: This Multiply can be avoided for "resizing by shape" right ?

File: ngraph_creator/operations/src/ResizeBilinear.cpp [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    // FLOAT16 type check added for future when VPUX plugin support is added
+    if (checkInputOperandType(1, (int32_t)OperandType::FLOAT32) ||
+        checkInputOperandType(1, (int32_t)OperandType::FLOAT16)) {
+        attrs.shape_calculation_mode = ngraph::op::v4::Interpolate::ShapeCalcMode::scales;
+        width_scale = sModelInfo->ParseOperationInput<float>(mNnapiOperationIndex, 1);
+        height_scale = sModelInfo->ParseOperationInput<float>(mNnapiOperationIndex, 2);
+    } else if (checkInputOperandType(1, (int32_t)OperandType::INT32)) {
+        attrs.shape_calculation_mode = ngraph::op::v4::Interpolate::ShapeCalcMode::scales;
```
Comment (akodanka) [inline ngraph_creator/operations/src/ResizeBilinear.cpp:98]: Can we not use ShapeCalcMode::sizes ?

File: ngraph_creator/operations/src/ResizeBilinear.cpp [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    std::vector<int32_t> axes_vec = {2, 3};
+    auto axesNode = createConstNode(ngraph::element::i32, {2}, axes_vec);
+
+    outputNode = std::make_shared<ngraph::op::v4::Interpolate>(inputNode, outputShapeNode,
+                                                               scaleNode, axesNode, attrs);
+    if (!useNchw) {
+        outputNode = transpose(NCHW_NHWC, outputNode);
+        mNgraphNodes->setForcedNchw(mDefaultOutputIndex, false);
```
Comment (vbenni) [inline ngraph_creator/operations/src/ResizeBilinear.cpp:129]: You may have to remove this too, the forcedNCHW like the inputnode.

Comment (Rjasuja) [reply ngraph_creator/operations/src/ResizeBilinear.cpp:129]: i removed it in the second commit

Comment (Rjasuja) [reply ngraph_creator/operations/src/ResizeBilinear.cpp:129]: Will update this. Thanks

**Comment by Nagamani71** [comment]
LGTM

File: ngraph_creator/operations/src/ResizeBilinear.cpp [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    switch (inputsSize) {
+        case 6:
+            half_pixel = sModelInfo->ParseOperationInput<uint8_t>(mNnapiOperationIndex, 5);
+        case 5:
+            align_corners = sModelInfo->ParseOperationInput<uint8_t>(mNnapiOperationIndex, 4);
+        case 4:
+            layout = sModelInfo->ParseOperationInput<uint8_t>(mNnapiOperationIndex, 3);
+        default:
+            break;
+    }
```
Comment (RavirajSitaram) [inline ngraph_creator/operations/src/ResizeBilinear.cpp:60]: This results in compiler warnings. Can u look into adding appropriate flags to remove the warnings.

Comment (Rjasuja) [reply ngraph_creator/operations/src/ResizeBilinear.cpp:63]: Updated code to remove warnings.

Comment (Rjasuja) [reply ngraph_creator/operations/src/ResizeBilinear.cpp:63]: Updated code to remove warnings

**Review by RavirajSitaram — APPROVED** [review]
LGTM

**Pull request merged**